sdl-lang/src/main/scala/io/smartdatalake/meta/configexporter/ConfigJsonExporter.scala

```diff
@@ -0,0 +1,85 @@
+      .text("Whether to add an additional property 'origin' including source filename and line number to first class configuration objects.")
+    help("help").text("Display the help text.")
+  }
+
+  /**
+   * Takes as input a SDL Config and exports it as one json document, everything resolved.
+   * Additionally a separate file with the mapping of first class config objects to source code origin is created.
+   */
+  def main(args: Array[String]): Unit = {
+    val config = ConfigJsonExporterConfig()
+    // Parse all command line arguments
+    parser.parse(args, config) match {
+      case Some(config) =>
+
+        // create json
+        val configAsJson = exportConfigJson(config)
+
+        // write file
+        logger.info(s"Writing config json to file ${config.filename}")
+        Files.write(Paths.get(config.filename), configAsJson.getBytes, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)
+
+      case None =>
+        logAndThrowException(s"Aborting ${appType} after error", new ConfigurationException("Couldn't set command line parameters correctly."))
+    }
+  }
+
+  def exportConfigJson(config: ConfigJsonExporterConfig): String = {
+    val defaultHadoopConf: Configuration = new Configuration()
+    val sdlConfig = ConfigLoader.loadConfigFromFilesystem(config.configPaths, defaultHadoopConf)
+    // remove additional config paths introduced by system properties...
+    val configKeysToRemove = sdlConfig.root.keySet().asScala.diff(Set(ConfigParser.CONFIG_SECTION_ACTIONS, ConfigParser.CONFIG_SECTION_CONNECTIONS, ConfigParser.CONFIG_SECTION_DATAOBJECTS, ConfigParser.CONFIG_SECTION_GLOBAL))
+    val reducedSdlConfig = configKeysToRemove.foldLeft(sdlConfig)((config,key) => config.withoutPath(key))
+    // enrich origin of first class config objects
+    val descriptionRegex = "(.*): ([0-9]+)(-[0-9]+)?".r
+    val enrichedSdlConfig = reducedSdlConfig.root().keySet().asScala.diff(Set(ConfigParser.CONFIG_SECTION_GLOBAL)).foldLeft(reducedSdlConfig){
+      case (config,sectionKey) =>
+        config.getConfig(sectionKey).root.keySet().asScala.foldLeft(config) {
+          case (config,objectKey) =>
+            val objectConfig = config.getConfig(s"$sectionKey.$objectKey")
+            // parse origins description, as we can not access the detailed private properties
+            val (path, lineNumber, endLineNumber) = objectConfig.origin.description match {
+              case descriptionRegex(path, lineNumber, endLineNumber) =>
+                (path, lineNumber.toInt, Option(endLineNumber).map(_.toInt))
+            }
+            val config1 = config
+              .withValue(s"$sectionKey.$objectKey.origin.path", ConfigValueFactory.fromAnyRef(path))
+              .withValue(s"$sectionKey.$objectKey.origin.lineNumber", ConfigValueFactory.fromAnyRef(lineNumber))
+            if (endLineNumber.isDefined) config1.withValue(s"$sectionKey.$objectKey.origin.endLineNumber", ConfigValueFactory.fromAnyRef(endLineNumber.get))
```

Comment:
FYI, endLineNumber is never included in the test file produced by the unit test

Reply:
True, the Hocon parser foresees this format, but doesnt fill endLineNumber for now...